### PR TITLE
[JS] feat: Who bot sample, add user sign in method for end user

### PR DESCRIPTION
**Pull request by singhk97**
#### work in progress
- [x] Update the sample to include more graph calls and test it

## Linked issues

closes: #890 #895

#### Change details

- Added Who bot sample
- Added `getTokenOrSignInUser` method in `Application` class. It is meant to be the single catch-all method to sign a user in and get the token. If the user is signed in it simply returns the token. If not, it triggers the sign in flow. Starting the sign in flow will only work if the incoming activity supports it (ex. `message`, `messageExtension/fetch`...etc). This method will work in action handlers as well.

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (we use [TypeDoc](https://typedoc.org/) to document our code)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

File: js/packages/teams-ai/src/Application.ts
Comment (singhk97) [inline js/packages/teams-ai/src/Application.ts:978]: @blackchoey this is meant to be the method the end-user calls if they want to trigger sign in. It will put the user in "sign in" state and the subsequent incomming activities will be routed to the authentication logic in the `Application.run()` until sign in flow completes - either with a success or failure.

File: js/samples/07.whoBot/src/bot.ts
Comment (blackchoey) [inline js/samples/07.whoBot/src/bot.ts:26]: Should we comment the OpenAI settings? It seems developers only need to config one of OpenAI/AzureOpenAI, configuring both might be confusing. We can keep the commentted OpenAI settings as a reference for developers.

File: js/samples/07.whoBot/src/bot.ts
Comment (blackchoey) [inline js/samples/07.whoBot/src/bot.ts:91]: Line 85 already gets the token, should we use the `token` variable?

**Comment by blackchoey** [comment]
FYI: not sure whether we should merge the changes to `main` branch

**Review by blackchoey — DISMISSED** [review]
Overall looks good to me. Left a few minor comments.

File: js/samples/07.whoBot/src/bot.ts
Comment (singhk97) [reply js/samples/07.whoBot/src/bot.ts:26]: We can keep both the settings so that the user can simply decide which one to use by setting the config value in the .env file. If they set `OPENAI_KEY` then open ai service will used. If they set `AZURE_OPENAI_KEY` and `AZURE_OPENAI_ENDPOINT` then azure open ai service will be used.

**Review by blackchoey — APPROVED** [review]
Approving first to avoid blocking the PR. But I see there're a lot of other files changed. Seems there should be no C# changes in this PR? Please check with @corinagum before merge to avoid bringing unexpected changes.

**Comment by singhk97** [comment]
The branch was based on `breaking-changes-2` which it shouldn't have. Creating another branch that's based on `main` instead. PR: https://github.com/microsoft/teams-ai/pull/911

**Pull request closed without merging**
